pkg/tsdb/azuremonitor/azuremonitor.go

```diff
@@ -298,6 +295,93 @@ func checkAzureMonitorResourceGraphHealth(dsInfo types.DatasourceInfo, subscript
 	return res, nil
 }
 
+func metricCheckHealth(dsInfo types.DatasourceInfo) (message string, defaultSubscription string, status backend.HealthStatus) {
+	defaultSubscription = dsInfo.Settings.SubscriptionId
+	metricsRes, err := queryMetricHealth(dsInfo)
+	if err != nil {
+		if ok := errors.Is(err, types.ErrorAzureHealthCheck); ok {
+			return fmt.Sprintf("Error connecting to Azure Monitor endpoint: %s", err.Error()), defaultSubscription, backend.HealthStatusError
+		}
+		return err.Error(), defaultSubscription, backend.HealthStatusError
+	}
+	defer func() {
+		err := metricsRes.Body.Close()
+		if err != nil {
+			message += err.Error()
+			status = backend.HealthStatusError
+		}
+	}()
+	if metricsRes.StatusCode != 200 {
+		body, err := io.ReadAll(metricsRes.Body)
+		if err != nil {
+			return err.Error(), defaultSubscription, backend.HealthStatusError
+		}
+		return fmt.Sprintf("Error connecting to Azure Monitor endpoint: %s", string(body)), defaultSubscription, backend.HealthStatusError
+	}
+	subscriptions, err := parseSubscriptions(metricsRes)
+	if err != nil {
+		return err.Error(), defaultSubscription, backend.HealthStatusError
+	}
+	if defaultSubscription == "" && len(subscriptions) > 0 {
+		defaultSubscription = subscriptions[0]
+	}
+
+	return "Successfully connected to Azure Monitor endpoint.", defaultSubscription, backend.HealthStatusOk
+}
+
+func logAnalytricsCheckHealth(dsInfo types.DatasourceInfo, defaultSubscription string) (message string, status backend.HealthStatus) {
+	logsRes, err := checkAzureLogAnalyticsHealth(dsInfo, defaultSubscription)
+	if err != nil {
+		if err.Error() == "no default workspace found" {
+			return "No Log Analytics workspaces found.", backend.HealthStatusUnknown
+		}
+		if ok := errors.Is(err, types.ErrorAzureHealthCheck); ok {
+			return fmt.Sprintf("Error connecting to Azure Log Analytics endpoint: %s", err.Error()), backend.HealthStatusUnknown
+		}
+		return err.Error(), backend.HealthStatusError
+	}
+	defer func() {
+		err := logsRes.Body.Close()
+		if err != nil {
+			message += err.Error()
+			status = backend.HealthStatusError
+		}
+	}()
+	if logsRes.StatusCode != 200 {
+		body, err := io.ReadAll(logsRes.Body)
+		if err != nil {
+			return err.Error(), backend.HealthStatusError
+		}
+		return fmt.Sprintf("Error connecting to Azure Log Analytics endpoint: %s", string(body)), backend.HealthStatusError
+	}
+	return "Successfully connected to Azure Log Analytics endpoint.", backend.HealthStatusOk
+}
+
+func graphLogHealthCheck(dsInfo types.DatasourceInfo, defaultSubscription string) (message string, status backend.HealthStatus) {
```

Comment:
I like the separation because `graphLogHealthCheck` interprets the results of the response (`checkAzureMonitorResourceGraphHealth`)

It seems like a logical divide and I mostly did this because of the analytics health check

Reply:
This is fair, I'm happy to go ahead with it 😊